src/sentry/tagstore/snuba/backend.py

```diff
@@ -433,6 +434,15 @@ def get_tag_keys_for_projects(
         max_unsampled_projects = _max_unsampled_projects
         # We want to disable FINAL in the snuba query to reduce load.
         optimize_kwargs = {"turbo": True}
+
+        # Add static sample amount to the query. Turbo will sample at 10% by
+        # default, but organizations with many events still get timeouts. A
+        # static sample creates more consistent performance.
+        organization_id = get_organization_id_from_project_ids(projects)
+        organization = Organization.objects.get_from_cache(id=organization_id)
+        if features.has("organizations:tag-key-sample-n", organization):
+            optimize_kwargs["sample"] = 1_000_000
```

Comment:
Thoughts on making this an option so you can tweak the sample size without a deploy?

Reply:
🤔 that seems like a good idea, although to be honest I'm planning to do research upfront about the right sample number, so I'm not expecting to change this a lot. Maybe I'm being too optimistic?

Reply:
I don't expect this to change often, but rather when you do need to change it, it's nice if we didn't have to wait hours to see the change reflected.

One other approach would be to use the admin tool to test various sample rates and just deciding based on that.

Reply:
Agreed! Added an option